src/code/actions.js

```diff
@@ -268,12 +271,42 @@ export function submitDrake(correctPhenotype, submittedPhenotype, correct, incor
   };
 }
 
-function _putEggInBasket(egg, basket, isCorrect) {
+function _rejectEggFromBasket(eggIndex, basketIndex) {
+  return {
+    type: actionTypes.EGG_REJECTED,
+    eggIndex,
+    basketIndex
+  };
+}
+
+export function rejectEggFromBasket() {
+  return (dispatch, getState) => {
+    const { submittedEggIndex, submittedBasketIndex } = getState();
+    dispatch(_rejectEggFromBasket(submittedEggIndex, submittedBasketIndex));
+  };
+}
+
+function _acceptEggInBasket(eggIndex, basketIndex) {
+  return {
+    type: actionTypes.EGG_ACCEPTED,
+    eggIndex,
+    basketIndex
+  };
+}
+
+export function acceptEggInBasket() {
+  return (dispatch, getState) => {
+    const { submittedEggIndex, submittedBasketIndex } = getState();
+    dispatch(_acceptEggInBasket(submittedEggIndex, submittedBasketIndex));
+  };
+}
+
+function _submitEggForBasket(eggIndex, basketIndex, isCorrect) {
```

Comment:
Here, and in `acceptEggInBasket`, `rejectEggInBasket`, and in the final state of the `baskets` array updated by the reducer, we only have a reference to an `eggIndex`. Thinking both to our ITS integration and the Paul Horwitz of the future trying to interpret logs, this index is fairly opaque because it doesn't directly refer to anything in the state object. I think in all other cases where we've passed around an index in an action, or referred to an index elsewhere in the state, we've explicitly referred to something you can look up in the state, like `drakes[i]`. (This makes these indexes rather like using normalized JSON)

A couple ways this could be more explicit: (1) pass the drake index directly, instead of the eggs index -- `eggs[0]` is basically `drakes[3]`, it seems; (2) for aid in ITS/research, passing the genotype of the drake, like we do in `submitDrake`.